website/src/views/mpe/MpeContainer.tsx

```diff
@@ -1,29 +1,110 @@
+import { useState } from 'react';
+import { useLocation, useHistory } from 'react-router-dom';
+import type { MpePreference } from 'types/mpe';
+import classnames from 'classnames';
+import Modal from 'views/components/Modal';
+import MpeFormContainer from './form/MpeFormContainer';
 import styles from './MpeContainer.scss';
+import {
+  getLoginState,
+  getSSOLink,
+  getMpePreferences,
+  updateMpePreferences,
+  MpeSessionExpiredError,
+} from '../../apis/mpe';
 
-type Props = {
-  placeholder: true; // Remove this when new props are added.
-};
+const MpeContainer: React.FC = () => {
+  const [isGettingSSOLink, setIsGettingSSOLink] = useState(false);
+  const [isSessionExpired, setIsSessionExpired] = useState(false);
+  const isLoggedIn = getLoginState(useLocation(), useHistory());
+
+  const onLogin = (): Promise<void> => {
+    setIsGettingSSOLink(true);
+    return getSSOLink()
+      .then((ssoLink) => {
+        window.location.href = ssoLink;
+      })
+      .finally(() => {
+        setIsGettingSSOLink(false);
+      });
+  };
+
+  const getPreferences = (): Promise<MpePreference[]> =>
+    new Promise((resolve, reject) => {
+      getMpePreferences()
+        .then((preferences) => resolve(preferences))
+        .catch((err) => {
+          if (err instanceof MpeSessionExpiredError) {
+            setIsSessionExpired(true);
+          }
+          return reject(err);
+        });
```

Comment:
As above, the `new Promise` is unnecessary 

```suggestion
    getMpePreferences()
      .catch((err) => {
        if (err instanceof MpeSessionExpiredError) {
          setIsSessionExpired(true);
        }
        throw err;
      });
```